zerver/lib/upload.py

```diff
@@ -897,22 +935,41 @@ def ensure_avatar_image(self, user_profile: UserProfile, is_medium: bool = False
 
     def upload_emoji_image(
         self, emoji_file: IO[bytes], emoji_file_name: str, user_profile: UserProfile
-    ) -> None:
+    ) -> bool:
         emoji_path = RealmEmoji.PATH_ID_TEMPLATE.format(
             realm_id=user_profile.realm_id,
             emoji_file_name=emoji_file_name,
         )
 
         image_data = emoji_file.read()
-        resized_image_data = resize_emoji(image_data)
+        resized_image_data, is_animated, still_image_data = resize_emoji(image_data)
         write_local_file("avatars", ".".join((emoji_path, "original")), image_data)
         write_local_file("avatars", emoji_path, resized_image_data)
-
-    def get_emoji_url(self, emoji_file_name: str, realm_id: int) -> str:
-        return os.path.join(
-            "/user_avatars",
-            RealmEmoji.PATH_ID_TEMPLATE.format(realm_id=realm_id, emoji_file_name=emoji_file_name),
-        )
+        if is_animated:
+            write_local_file(
+                "avatars", emoji_path.replace(".gif", "-still.png"), cast(bytes, still_image_data)
+            )
+        return is_animated
+
+    def get_emoji_url(self, emoji_file_name: str, realm_id: int, still: bool = False) -> str:
+        if still:
+            if not emoji_file_name.endswith(".gif"):
+                return ""
```

Comment:
Why do we have this `still` check for returning `""` here, and not a variant in the similar other code path?

Reply:
Yeah, it's better to `assert`!